### PR TITLE
ar71xx: add support for Gainstrong MiniBox V3.2

### PR DESCRIPTION
**Pull request by p4u**
Adds MiniBox V3.2 ar71xx target (QCA99531 + QCA9887)

- Info http://oolite.cn/minibox-v3-2-qca9531-qca9887-module-pin-specification.html
- Reference patch: http://oolite.cn/wp-content/uploads/2017/09/minibox_v3_cc15.05_20160325.patch
- Installation: Upload firmware to uboot HTTP at 192.168.1.1 (press button for 5 seconds)

More info: https://forum.lede-project.org/t/adding-support-for-minibox-v3-2/8863

### CODE REVIEW

File: target/linux/ar71xx/base-files/lib/ar71xx.sh [hunk 1]
```diff
@@ -287,11 +287,17 @@ tplink_board_detect() {
 	"360000"*)
 		model="TP-Link TL-WDR3600"
 		;;
-	"3C0001"*)
+	"3c0001"*)
 		model="OOLITE"
+
```
Comment (blogic) [inline target/linux/ar71xx/base-files/lib/ar71xx.sh]: no need to add a blank line here

File: target/linux/ar71xx/base-files/lib/upgrade/platform.sh [hunk 2]
```diff
@@ -460,6 +461,7 @@ platform_check_image() {
 	tl-wr941nd|\
 	tl-wr941nd-v5|\
 	tl-wr941nd-v6)
+
```
Comment (blogic) [inline target/linux/ar71xx/base-files/lib/upgrade/platform.sh]: no need to add a blank line here

File: target/linux/ar71xx/base-files/lib/ar71xx.sh [hunk 3]
```diff
@@ -287,11 +287,17 @@ tplink_board_detect() {
...
+		[ "$hwid" = "3c00010c" -a "$mid" = "00000001" ] && {
+			model="MINIBOX_V3.2"
+			hwver=""
```
Comment (blogic) [inline target/linux/ar71xx/base-files/lib/ar71xx.sh]: should you not set this to v3.2 ?

File: target/linux/ar71xx/base-files/lib/ar71xx.sh [hunk 4]
```diff
@@ -287,11 +287,17 @@ tplink_board_detect() {
...
+		}
 		;;
-	"3C0002"*)
+	"3c0002"*)
 		model="MINIBOX_V1"
+		hwver=""
```
Comment (blogic) [inline target/linux/ar71xx/base-files/lib/ar71xx.sh]: and this to v1 ?

File: target/linux/ar71xx/files/arch/mips/ath79/Makefile [hunk 5]
```diff
@@ -126,6 +126,7 @@ obj-$(CONFIG_ATH79_MACH_GL_INET)		+= mach-gl-inet.o
 obj-$(CONFIG_ATH79_MACH_GL_MIFI)		+= mach-gl-mifi.o
 obj-$(CONFIG_ATH79_MACH_GL_USB150)		+= mach-gl-usb150.o
 obj-$(CONFIG_ATH79_MACH_GS_MINIBOX_V1)		+= mach-gs-minibox-v1.o
+obj-$(CONFIG_ATH79_MACH_GS_MINIBOX_V32)		+= mach-gs-minibox-v32.o
```
Comment (blogic) [inline target/linux/ar71xx/files/arch/mips/ath79/Makefile]: extra tab, remove it please

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt [hunk 6]
```diff
@@ -837,6 +837,18 @@ config ATH79_MACH_GS_MINIBOX_V1
 	select ATH79_DEV_USB
 	select ATH79_DEV_WMAC
 
+config ATH79_MACH_GS_MINIBOX_V32
```
Comment (blogic) [inline target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: this whole block is not properly indented

File: target/linux/ar71xx/files/arch/mips/ath79/mach-gs-minibox-v32.c [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+#include "machtypes.h"
+#include "pci.h"
+
+#define GS_MINIBOX_V3_GPIO_LED_STATUS		14
+#define GS_MINIBOX_V3_GPIO_BTN_RST                 17
+#define GS_MINIBOX_V3_KEYS_POLL_INTERVAL           20	/* msecs */
+#define GS_MINIBOX_V3_KEYS_DEBOUNCE_INTERVAL	(3 * GS_MINIBOX_V3_KEYS_POLL_INTERVAL)
+#define GS_MINIBOX_V3_MAC0_OFFSET                  0
+#define GS_MINIBOX_V3_MAC1_OFFSET                  6
+#define GS_MINIBOX_V3_WMAC_CALDATA_OFFSET          0x1000
```
Comment (blogic) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-gs-minibox-v32.c]: all the defines above should use tabs and not spaces as separators

File: target/linux/ar71xx/base-files/lib/ar71xx.sh (hunk 1 above)
Comment (p4u) [reply target/linux/ar71xx/base-files/lib/ar71xx.sh]: Fine.

File: target/linux/ar71xx/base-files/lib/ar71xx.sh (hunk 4 above)
Comment (p4u) [reply target/linux/ar71xx/base-files/lib/ar71xx.sh]: Yes, hwver on V1 was missing so I added it for coherence (someone on #lede-dev told me to do so)

File: target/linux/ar71xx/base-files/lib/upgrade/platform.sh (hunk 2 above)
Comment (p4u) [reply target/linux/ar71xx/base-files/lib/upgrade/platform.sh]: right

File: target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt (hunk 6 above)
Comment (p4u) [reply target/linux/ar71xx/files/arch/mips/ath79/Kconfig.openwrt]: right

File: target/linux/ar71xx/files/arch/mips/ath79/mach-gs-minibox-v32.c (hunk 7 above)
Comment (p4u) [reply target/linux/ar71xx/files/arch/mips/ath79/mach-gs-minibox-v32.c]: Sorry for messing tabs and spaces, I'll resend patch

File: target/linux/ar71xx/base-files/lib/ar71xx.sh (hunk 3 above)
Comment (p4u) [reply target/linux/ar71xx/base-files/lib/ar71xx.sh]: I'm following the same rules of Minibox v1
https://github.com/p4u/source/blob/dc32cae3cc918719208502bad07f8075f6990e38/target/linux/ar71xx/base-files/lib/ar71xx.sh#L299

**Comment by p4u** [comment]
Changes done at new PR https://github.com/lede-project/source/pull/1577

**Comment by blogic** [comment]
With the remerge in progress, all PRs on the lede-project organisation will be closed. Please help getting this merged or rebase/post it on the openwrt project page (https://github.com/openwrt/openwrt/pulls). All remaining PRs will be closed in 30 days.

**Comment by pepe2k** [comment]
Superseded by https://github.com/lede-project/source/pull/1578

**Pull request closed without merging**